File: PS2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int main(void)
{
    //prompt for text
    string text = get_string("Text: ");


    //get letters
    //set variable for total letters to 0
    int letters = 0;

    //find total length of string
    int len = strlen(text);

    //count each letter and store in varaible 'letters'
    for (int i = 0; i < len; i++)
    {
        if (text[i] > 64 && text[i] < 91)
        {
            letters++;
        }
        if (text[i] > 96 && text[i] < 123)
        {
            letters++;
        }
    }


    //find number of words
    int words = 1;

    for (int w = 1; w < len; w++)
    {
        if (text[w] == 32)
        {
            words++;
        }
    }


    //find numer of sentences
    int sentences = 0;

    for (int s = 0; s < len; s++)
    {
        if ((text[s] == 33 || text[s] == 46 || text[s] == 63))
        {
            sentences++;
        }
    }


    //putting it all together
    //find value of L
    float L = 0;
    L = (((float) letters) / ((float) words) * 100);

    //find value of S
    float S = 0;
    S = (((float) sentences) / ((float) words) * 100);

    //equation and round
    float index = 0;
    index = 0.0588 * L - 0.296 * S - 15.8;

    int grade = round(index);

    if (grade > 16)
    {
        printf("Grade 16+\n");
    }

    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }

    else
    {
        printf("Grade %i\n", grade);
    }
}

File: pset6/readability.py
from cs50 import get_string
import re

# prompt user for input
text = get_string("Text: ")

# set number of letters to 0
letters = 0

# initialize text length
length = len(text)

# initialize number of words
words = 0

# initialize number of sentences
sentences = 0

# count each letter
i = 0
while i < length:
    if text[i].isalpha():
        letters += 1

    if (i == 0 and text[i] != " ") or (i != length and text[i] == " " and text[i + 1] != " "):
        words += 1

    if text[i] == "." or text[i] == "!" or text[i] == "?":
        sentences += 1
    i += 1

# average number of letters per 100 words
L = (letters / words) * 100

# average number of sentences per 100 words
S = (sentences / words) * 100

# Coleman-Liau index
index = round(0.0588 * L - 0.296 * S - 15.8)

if index < 1:
    print("Before Grade 1")

elif index >= 16:
    print("Grade 16+")

else:
    print(f"Grade {index}")
    
